src/qibo/tensorflow/cgates.py

```diff
@@ -797,5 +802,100 @@ def __init__(self, q: int, px: float = 0, py: float = 0, pz: float = 0,
 
     @staticmethod
     def _invert(gate):
-        """For Pauli gates we can use same gate for state inversion for efficiency."""
+        # for Pauli gates we can use same gate as inverse for efficiency
         return gate
+
+
+class ResetChannel(UnitaryChannel, base_gates.ResetChannel):
+
+    def __init__(self, q: int, p0: float = 0.0, p1: float = 0.0,
+                 seed: Optional[int] = None):
+        TensorflowGate.__init__(self)
+        base_gates.ResetChannel.__init__(self, q, p0=p0, p1=p1, seed=seed)
+        self.inv_gates = tuple()
+
+    @staticmethod
+    def _invert(gate):
+        if isinstance(gate, base_gates.Collapse):
+            return None
+        return gate
+
+    def _state_vector_call(self, state: tf.Tensor) -> tf.Tensor:
+        self.module.TensorflowGate._set_nqubits(self, state)
+        not_collapsed = True
+        if np.random.random() < self.probs[-2]:
+            state = self.gates[-2](state)
+            not_collapsed = False
+        if np.random.random() < self.probs[-1]:
+            if not_collapsed:
+                state = self.gates[-2](state)
+            state = self.gates[-1](state)
+        return state
+
+
+class ThermalRelaxationChannel(TensorflowGate, base_gates.ThermalRelaxationChannel):
+
+    def __new__(cls, q, t1, t2, time, excited_population=0, seed=None):
+        if BACKEND.get('GATES') == "custom":
+            cls_a = _ThermalRelaxationChannelA
+            cls_b = _ThermalRelaxationChannelB
+        else:
+            from qibo.tensorflow import gates
+            cls_a = gates._ThermalRelaxationChannelA
+            cls_b = gates._ThermalRelaxationChannelB
+        if t2 > t1:
```

Comment:
Ok, I suppose checking if t1,t2 are doubles in in `base_gates.ThermalRelaxationChannel` is probably useless at this stage.